New Functions for NPC, Objects, GroundItems, and InventoryItems

michaelmob opened the pull request:
New Functions for NPC, Objects, and GroundItems
- .IsOnMS
- .RotateTo
- .WalkTo
- .Hover
- .Click
- .ChooseOption

New Functions for GroundItems
- .Take

New Functions for Tile
- .TileOnMS (Tile on MainScreen)
- .TileRandom (Get random tile near specified tile)
  <a href='#crh-start'></a><a href='#crh-data-%7B%22processed%22%3A%20%5B%22https%3A//github.com/Elfyyy/OSR-Reflection/pull/58%23issuecomment-85792753%22%5D%2C%20%22comments%22%3A%20%7B%22General%20Comment%22%3A%20%7B%22html_url%22%3A%20%22https%3A//github.com/Elfyyy/OSR-Reflection/pull/58%23issuecomment-85792753%22%2C%20%22comments%22%3A%20%5B%7B%22body%22%3A%20%22%40Elfyyy%20what%27s%20happening%20with%20this%3F%20I%20think%20some%20of%20these%20would%20be%20useful.%20If%20you%20have%20any%20issues%20you%20should%20post%20them%20here%20so%20they%20can%20be%20fixed%22%2C%20%22created_at%22%3A%20%222015-03-25T02%3A32%3A00Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/8165365%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/SRLTheMayor%22%7D%7D%5D%2C%20%22title%22%3A%20%22General%20Comment%22%7D%7D%7D'></a>
  <a href='https://www.codereviewhub.com/'><img src='http://www.codereviewhub.com/site/github-bar.png' height=40></a>
- [ ] <a href='#crh-comment-General Comment'></a> <img src='http://www.codereviewhub.com/site/github-remaining.png' height=16 width=60>&nbsp;<b><a href='https://github.com/Elfyyy/OSR-Reflection/pull/58#issuecomment-85792753'>General Comment</a></b>
- <a href='https://github.com/SRLTheMayor'><img border=0 src='https://avatars.githubusercontent.com/u/8165365?v=3' height=16 width=16'></a> @Elfyyy what's happening with this? I think some of these would be useful. If you have any issues you should post them here so they can be fixed

<a href='https://www.codereviewhub.com/Elfyyy/OSR-Reflection/pull/58?mark_as_completed=1'><img src='http://www.codereviewhub.com/site/github-mark-as-completed.png' height=26></a>&nbsp;<a href='https://www.codereviewhub.com/Elfyyy/OSR-Reflection/pull/58?approve=1'><img src='http://www.codereviewhub.com/site/github-approve.png' height=26></a>&nbsp;<a href='https://github.com/Elfyyy/OSR-Reflection/pull/58'><img src='http://www.codereviewhub.com/site/github-refresh.png' height=26></a>
<a href='#crh-end'></a>

Commit:
New Functions for NPC, Objects, and GroundItems

New Functions for NPC, Objects, and GroundItems
* .IsOnMS
* .RotateTo
* .WalkTo
* .Hover
* .Click
* .ChooseOption

New Functions for GroundItems
* .Take

New Functions for Tile
* .TileOnMS (Tile on MainScreen)
* .TileRandom (Get random tile near specified tile)

Commit:
Fix Impossible Overload Errors

Elfyyy commented:
This pull request is now managed with CodeReviewHub. <a href='https://github.com/Elfyyy/OSR-Reflection/pull/58'>Refresh</a>.<a href='#crh-update-none'></a>

Commit:
Removed Overloads for Optional Parameters

Removed Overloads for Optional Parameters
Added Item's name for Take for better accuracy of picking the correct
item up

Commit:
New Functions for Inventory Items

New Functions for Inventory Items
* .Hover
* .Click
* .ChooseOption
* .Use
* .Drop (changed to use .ChooseOption)

Commit:
Fix for .ChooseOption not exiting when left-click

SRLTheMayor commented:
@Elfyyy what's happening with this? I think some of these would be useful. If you have any issues you should post them here so they can be fixed

Commit:
Merge remote-tracking branch 'upstream/master'
